### ISSUE TITLE
`authStore` not saving sate on React Native.

### ISSUE THREAD

**Issue by BenMcAvoy** [pocketbase/pocketbase#5013]
I have the following code to handle a login form being submitted:
```tsx
const onSubmit: SubmitHandler<FormValues> = async (values) => {
	login({ email: values.email, pass: values.password });

	while (!pb.authStore.isValid) {
		console.log("Waiting for login");
	}

	// await waitFor(() => pb.authStore.isValid);

	console.log("Logged in");

	reset();
	router.replace("/");
}
```

(Screenshot from dev tools connected to my phone running the app)
![image](https://github.com/pocketbase/pocketbase/assets/143108602/9c98b20a-6ec7-4a80-a8ee-9df6aef4521c)

The `authStore.isValid` is never set to true. If I do this in the web on the other hand using Expo Web, it successfully validates. I can tell it is actually making the web request because in the network tab it is returning with a 200 status code and a token is provided.

**Comment by ganigeorgiev** [comment]
This is not a PocketBase issue.

It was recently reported in https://github.com/pocketbase/pocketbase/discussions/4787#discussioncomment-9575109.

TLDR; React Native 0.17+ and Expo 51 ships with native `atob` function that behaves differently compared to the browser implementation. As mentioned in [one of my replies](https://github.com/pocketbase/pocketbase/discussions/4787#discussioncomment-9577378) this is already fixed in the hermes repo but waits to be chery-picked for release - https://github.com/reactwg/react-native-releases/issues/287.

For now the options are to downgrade until the above fix get released. If the RN team ended up taking too long to merge it I'll consider applying a workaround from our end.

**Comment by BenMcAvoy** [comment]
Thank you for the quick reply.

How can I go about solving/working around this this then? I have changed my Expo version to `v50` and React Native to `0.73.8` but now Expo Go (I don't own a Macbook). Simply changing to React Native 0.73.8 did not help and changing to v50 made it no longer work with Expo Go.

If not, could you advise me on how to migrate to the hermes branch with the patch applied?

**Comment by ganigeorgiev** [comment]
I've pushed a minor [JS SDK v0.21.3 release](https://github.com/pocketbase/js-sdk/releases/tag/v0.21.3) that will temporary enforce the `atob` polyfill when using React Native. This should "fix" `pb.authStore.isValid` to report the correct value under Android and iOS.
